# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1285711932 14400
#      Tue Sep 28 18:12:12 2010 -0400
# Node ID e7864673c31fa552612150d4ce1c28afef04ae87
# Parent  169f59f626d34426369d0c175c7e34cb66a1c863
new Cell (std::list<std::string>&) constructor

diff --git a/src/Cell.cc b/src/Cell.cc
--- a/src/Cell.cc
+++ b/src/Cell.cc
@@ -58,16 +58,35 @@ Cell::Cell (const string_vector& sv, boo
               s = (pos == std::string::npos) ? "" : s.substr (0, pos+1);
             }
 
           elem(i,0) = s;
         }
     }
 }
 
+Cell::Cell (const std::list<std::string>& lst)
+  : Array<octave_value> ()
+{
+  size_t n = lst.size ();
+
+  if (n > 0)
+    {
+      resize (dim_vector (n, 1));
+
+      octave_idx_type i = 0;
+
+      for (std::list<std::string>::const_iterator it = lst.begin ();
+           it != lst.end (); it++)
+        {
+          elem(i++,0) = *it;
+        }
+    }
+}
+
 Cell::Cell (const Array<std::string>& sa)
   : Array<octave_value> (sa.dims ())
 {
   octave_idx_type n = sa.numel ();
 
   octave_value *dst = fortran_vec ();
   const std::string *src = sa.data ();
 
diff --git a/src/Cell.h b/src/Cell.h
--- a/src/Cell.h
+++ b/src/Cell.h
@@ -58,16 +58,18 @@ public:
   Cell (const Array<octave_value>& c)
     : Array<octave_value> (c) { }
 
   Cell (const Array<octave_value>& c, octave_idx_type nr, octave_idx_type nc)
     : Array<octave_value> (c, dim_vector (nr, nc)) { }
 
   Cell (const string_vector& sv, bool trim = false);
 
+  Cell (const std::list<std::string>& lst);
+
   Cell (const Array<std::string>& sa);
 
   Cell (const dim_vector& dv, const string_vector& sv, bool trim = false);
 
   Cell (const Cell& c)
     : Array<octave_value> (c) { }
 
   bool is_cellstr (void) const;
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-09-28  John W. Eaton  <jwe@octave.org>
+
+	* Cell.h, Cell.cc (Cell::Cell (const std::list<std::string>&)):
+	New constructor.
+
 2010-09-28  John P. Swensen  <jpswensen@gmail.com>
 
 	* DLD-FUNCTIONS/__magick_read__.cc (maybe_initialize_magick):
 	Warn on initialization of *Magick if QuantumDepth is less than 32.
 
 2010-09-28  John P. Swensen  <jpswensen@gmail.com>
 
 	* DLD-FUNCTIONS/__magick_read__.cc (F__magick_format_list__):
